payments-core/src/main/java/com/stripe/android/paymentsheet/injection/PaymentSheetViewModelModule.kt

```diff
@@ -55,53 +29,6 @@ internal abstract class PaymentSheetViewModelModule {
             return starterArgs.clientSecret
         }
 
-        @Provides
-        @IOContext
-        fun provideWorkContext(): CoroutineContext = Dispatchers.IO
-
-        @Provides
-        @Singleton
-        fun provideApiRequestOptions(
-            lazyPaymentConfiguration: Lazy<PaymentConfiguration>
-        ) = ApiRequest.Options(
-            apiKey = lazyPaymentConfiguration.get().publishableKey,
-            stripeAccount = lazyPaymentConfiguration.get().stripeAccountId
-        )
```

Comment:
`ApiRequest.Options` is a generic type, different API calls need different parameters. Providing this here would get the `PaymentConfiguration` immediately, which was fine because only the `PaymentSheetViewModel` used it, and it's instantiated later in the flow. Now that other classes also need the options, it's better to pass the `Lazy<PaymentConfiguration>` as parameter and let the classes create their own `ApiRequest.Options` from it.

Reply:
another option is don't inject it as `@Singleton`, and use `Provider<ApiRequest.Options>`, so you create a new one with latest `PaymentConfiguration` whenever needed.

Reply:
As I mentioned, different API calls need different `ApiRequest.Options`. `CustomerApiRepository` for example uses the customer's ephemeral key as `apiKey`. So to inject `ApiRequest.Options` we'd need to annotate and provide multiple instances of it. I think injecting the `Lazy<PaymentConfiguration>` and letting the class fetch what it needs from it is simpler.